python-backend/app/api/permits/permit_amendment/models/permit_amendment.py

```diff
@@ -72,29 +102,26 @@ def validate_permit_id(self, key, permit_id):
 
     @validates('received_date')
     def validate_received_date(self, key, received_date):
-        if not received_date:
-            return received_date
-        if received_date.isoformat() == '9999-12-31':
-            raise AssertionError(
-                'Permit amendment received date should be set to null if not known.')
-        if received_date > datetime.today():
-            raise AssertionError('Permit amendment received date cannot be set to the future.')
+        if received_date:
+            if received_date.isoformat() == '9999-12-31':
```

Comment:
What's the reason for changing this from the early return style?

Reply:
Just how we have talked about the pattern, early positive returns can be harder to manage as function size grows, this instance was probably fine, but I had already done it before the catch was implemented this way, so I kept my version.

Reply:
Where/when was this pattern discussed? I favour an early return when exiting due to an error.

It says "we don't have a received date, so there's nothing else to do here. Exiting."

This, to me, seems clearer.

Reply:
To clarify, i think we call agree that returning early on an error makes sense and it totally fine, we do it lots in the resources. But I prefer to have happy path always make it to the end of the function.

Reply:
Ahh.. perfect. Thanks for clarifying